llvm/include/llvm/Support/PropertySetIO.h

```diff
@@ -201,11 +201,21 @@ class PropertySetRegistry {
       PropSet.insert(std::make_pair(Prop.first, PropertyValue(Prop.second)));
   }
 
+  template <typename T>
```

Comment:
- Please add a comment when this function should be used instead of `void add(StringRef Category, const std::map<StringRef, T> &Props)` above. I.e. when the order of properties matters.
- Suggest to make `add` templated on the map type to have single implementation for both `std::map<StringRef, T>` and `std::map<StringRef, T>`